File: model/src/main/java/io/sparkled/model/entity/Sequence.java
package io.sparkled.model.entity;

import javax.persistence.*;
import java.util.Objects;

@Entity
@Table(name = "sequence")
public class Sequence {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Integer id;

    @Basic
    @Column(name = "stage_id")
    private Integer stageId;

    @Basic
    @Column(name = "name")
    private String name;

    @Basic
    @Column(name = "artist")
    private String artist;

    @Basic
    @Column(name = "album")
    private String album;

    @Basic
    @Column(name = "duration_frames")
    private Integer durationFrames;

    @Basic
    @Column(name = "frames_per_second")
    private Integer framesPerSecond = 60;

    @Enumerated(EnumType.STRING)
    @Column(name = "status")
    private SequenceStatus status;

    public Integer getId() {
        return id;
    }

    public Sequence setId(Integer id) {
        this.id = id;
        return this;
    }

    public Integer getStageId() {
        return stageId;
    }

    public Sequence setStageId(Integer stageId) {
        this.stageId = stageId;
        return this;
    }

    public String getName() {
        return name;
    }

    public Sequence setName(String name) {
        this.name = name;
        return this;
    }

    public String getArtist() {
        return artist;
    }

    public Sequence setArtist(String artist) {
        this.artist = artist;
        return this;
    }

    public String getAlbum() {
        return album;
    }

    public Sequence setAlbum(String album) {
        this.album = album;
        return this;
    }

    public Integer getDurationFrames() {
        return durationFrames;
    }

    public Sequence setDurationFrames(Integer durationFrames) {
        this.durationFrames = durationFrames;
        return this;
    }

    public Integer getFramesPerSecond() {
        return framesPerSecond;
    }

    public Sequence setFramesPerSecond(Integer framesPerSecond) {
        this.framesPerSecond = framesPerSecond;
        return this;
    }

    public SequenceStatus getStatus() {
        return status;
    }

    public Sequence setStatus(SequenceStatus status) {
        this.status = status;
        return this;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Sequence sequence = (Sequence) o;
        return Objects.equals(id, sequence.id) &&
                Objects.equals(stageId, sequence.stageId) &&
                Objects.equals(name, sequence.name) &&
                Objects.equals(artist, sequence.artist) &&
                Objects.equals(album, sequence.album) &&
                Objects.equals(durationFrames, sequence.durationFrames) &&
                Objects.equals(framesPerSecond, sequence.framesPerSecond) &&
                status == sequence.status;
    }

    @Override
    public int hashCode() {
        return Objects.hash(id, stageId, name, artist, album, durationFrames, framesPerSecond, status);
    }

    @Override
    public String toString() {
        return "Sequence{" +
                "id=" + id +
                ", stageId=" + stageId +
                ", name='" + name + '\'' +
                ", artist='" + artist + '\'' +
                ", album='" + album + '\'' +
                ", durationFrames=" + durationFrames +
                ", framesPerSecond=" + framesPerSecond +
                ", status=" + status +
                '}';
    }
}

File: model/src/main/kotlin/io/sparkled/model/entity/Sequence.kt
package io.sparkled.model.entity

import javax.persistence.Basic
import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.EnumType
import javax.persistence.Enumerated
import javax.persistence.GeneratedValue
import javax.persistence.GenerationType
import javax.persistence.Id
import javax.persistence.Table

@Entity
@Table(name = "sequence")
class Sequence {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private var id: Int? = null

    @Basic
    @Column(name = "song_id")
    private var songId: Int? = null

    @Basic
    @Column(name = "stage_id")
    private var stageId: Int? = null

    @Basic
    @Column(name = "name")
    private var name: String? = null

    @Basic
    @Column(name = "frames_per_second")
    private var framesPerSecond: Int? = null

    @Enumerated(EnumType.STRING)
    @Column(name = "status")
    private var status: SequenceStatus? = null

    fun getId(): Int? {
        return id
    }

    fun setId(id: Int?): Sequence {
        this.id = id
        return this
    }

    fun getSongId(): Int? {
        return songId
    }

    fun setSongId(songId: Int?): Sequence {
        this.songId = songId
        return this
    }

    fun getStageId(): Int? {
        return stageId
    }

    fun setStageId(stageId: Int?): Sequence {
        this.stageId = stageId
        return this
    }

    fun getName(): String? {
        return name
    }

    fun setName(name: String?): Sequence {
        this.name = name
        return this
    }

    fun getFramesPerSecond(): Int? {
        return framesPerSecond
    }

    fun setFramesPerSecond(framesPerSecond: Int?): Sequence {
        this.framesPerSecond = framesPerSecond
        return this
    }

    fun getStatus(): SequenceStatus? {
        return status
    }

    fun setStatus(status: SequenceStatus?): Sequence {
        this.status = status
        return this
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as Sequence

        if (id != other.id) return false
        if (songId != other.songId) return false
        if (stageId != other.stageId) return false
        if (name != other.name) return false
        if (framesPerSecond != other.framesPerSecond) return false
        if (status != other.status) return false

        return true
    }

    override fun hashCode(): Int {
        var result = id ?: 0
        result = 31 * result + (songId ?: 0)
        result = 31 * result + (stageId ?: 0)
        result = 31 * result + (name?.hashCode() ?: 0)
        result = 31 * result + (framesPerSecond ?: 0)
        result = 31 * result + (status?.hashCode() ?: 0)
        return result
    }

    override fun toString(): String {
        return "Sequence(id=$id, songId=$songId, stageId=$stageId, name=$name, framesPerSecond=$framesPerSecond, status=$status)"
    }
}
